src/relations/async_replication.py

```diff
@@ -116,9 +120,16 @@ def __init__(self, charm: "PostgresqlOperatorCharm"):
     @property
     def _unit_ip(self) -> str:
         """Return this unit IP address for the replication relation."""
+        if not self._relation:
+            raise AsyncReplicationError("No relation to get IP for")
+
         if self._relation.name == REPLICATION_OFFER_RELATION:
-            return self.charm._replication_offer_ip
-        return self.charm._replication_consumer_ip
+            ip = self.charm._replication_offer_ip
+        ip = self.charm._replication_consumer_ip
```

Comment:
The new code always returns the value from `self.charm._replication_consumer_ip` (ignoring the value from `self.charm._replication_offer_ip`).